### PR TITLE
Fix virtualbox build and reflector installation, as well as reflector-init.service

### PR DESCRIPTION
**Pull request by juadde**
I wanted to investigate why reflector-init was not working with the latest boxes, and found that building the box failed. Pacman was failing to install reflector due to the missing --noconfirm, and virtualbox guest modules are now part of "linux" package directly, so there's no need to add them. I will investigate on the refector-init issue, it seems that for me reflector is trying to reach the mirror page before having DNS available.

### CODE REVIEW

File: http/install.sh [hunk 1]
```diff
@@ -28,7 +28,7 @@ mount "${device}2" /mnt
 if [ -n "${MIRROR}" ]; then
   echo "Server = ${MIRROR}" >/etc/pacman.d/mirrorlist
 else
-  pacman -Sy reflector
+  pacman -Sy --noconfirm reflector
```
Comment (shibumi) [inline http/install.sh:31]: This looks good to me

File: provision/virtualbox.sh [hunk 2]
```diff
@@ -3,5 +3,5 @@
 set -e
 set -x
 
-sudo pacman -S --noconfirm virtualbox-guest-utils-nox virtualbox-guest-modules-arch
+sudo pacman -S --noconfirm virtualbox-guest-utils-nox
```
Comment (shibumi) [inline provision/virtualbox.sh:6]: why did you remove virtualbox-guest-modules-arch? it has nothing to do with the issue you've mentioned

Comment (juadde) [reply provision/virtualbox.sh:6]: virtualbox-guest-modules-arch is not available in the archlinux repos, and the build failed with a "no target found" on pacman (however it is part of linux 5.6, that is obviously installed). So it is part of the "Fix virtualbox build". With this PR I was only fixing the build for virtualbox, no functional update. I will do another PR that fixes reflector-init service by waiting for network-online (instead of network) and before sshd is available, so if somebody uses the box with a pacman install at the begining of his Vagrantfile he has the right mirrorlist.

**Comment by shibumi** [comment]
lgtm

**Comment by n4nn31355** [comment]
Can this changes be released? The latest box version(v2020.05.06) is still some kind of broken.

**Comment by shibumi** [comment]
Oh yes, I still need to make a new release + trigger a new build for the boxes.
